week4/assignment/src/components/common/Button.tsx

```diff
@@ -0,0 +1,42 @@
+import styled from '@emotion/styled';
+import { Theme } from '../../styles/theme';
+
+interface ButtonProps {
+  type: 'button' | 'submit'
+  onClick?: () => void;
+  disabled?: boolean;
+  children: React.ReactNode;
+}
```

Comment:
```suggestion
interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  children: React.ReactNode;
}
```

요렇게 버튼 엘리먼트의 기본속성을 상속받으면 button이 가지고 있는 속성들을 명시하지 않고 사용할 수 있습니다!